### PR TITLE
BAH-4641 [Form2-Controls/ Composite components] Migrate Section to Carbon Accordion

### PR DESCRIPTION
**Pull request by lingeswaranTW**

### CODE REVIEW

File: src/components/bahmni-design-system/Section.jsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+      metadata: { label },
+      onEventTrigger,
+      patientUuid,
+      validate,
+      validateForm,
+      showNotification,
+    } = this.props;
+
+    const childProps = {
+      collapse,
```
Comment (ravinderkabli) [inline src/components/bahmni-design-system/Section.jsx:62]: blocking: `collapse` here comes from `this.props`, not `this.state`.

After the user clicks the accordion heading, `_onCollapse` updates `this.state.collapse` but `this.props.collapse` remains unchanged — so child controls receive the stale prop value instead of the current toggle state. `ObsGroupControl` (line 81) uses `this.state.collapse` in `childProps` for exactly this reason.

Fix:
```js
const childProps = {
  collapse: this.state.collapse,
  ...
};
```

File: src/components/bahmni-design-system/CarbonContainer.jsx [hunk 2]
```diff
@@ -13,6 +14,7 @@ import { FreeTextAutoComplete } from 'components/bahmni-design-system/FreeTextAu
 const carbonComponents = {
   text: TextBox,
   obsgroupcontrol: ObsGroupControl,
+  section: Section,
```
Comment (ravinderkabli) [inline src/components/bahmni-design-system/CarbonContainer.jsx]: blocking: `Section` (the raw class) is registered here, but `Section.render()` calls `this.props.intl.formatMessage(...)` — which requires the component to be wrapped with `injectIntl`. Without it, `this.props.intl` is `undefined` and every render will throw:

> TypeError: Cannot read properties of undefined (reading 'formatMessage')

Unlike `ObsGroupControl`, which only *spreads* `intl` onto `<IntlProvider>` (a no-op if undefined), `Section` calls methods on it directly — so this is a hard crash specific to Section.

Fix:
```js
import { SectionWithIntl } from 'components/bahmni-design-system/Section';
const carbonComponents = { section: SectionWithIntl, ... };
```

File: src/components/bahmni-design-system/Section.jsx [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    this.state = { errors: [], collapse };
+    this.onChange = this.onChange.bind(this);
+    this._onCollapse = this._onCollapse.bind(this);
+    this.onControlAdd = this.onControlAdd.bind(this);
+    this.onControlRemove = this.onControlRemove.bind(this);
+    this.onAddControl = this.onAddControl.bind(this);
+    this.onRemoveControl = this.onRemoveControl.bind(this);
+  }
+
+  componentDidUpdate(prevProps, prevState) {
```
Comment (ravinderkabli) [inline src/components/bahmni-design-system/Section.jsx]: suggestion: The second condition (`this.props.collapse !== prevState.collapse && prevProps.collapse === this.props.collapse`) diverges from `ObsGroupControl.componentDidUpdate`, which only checks `this.props.collapse !== prevProps.collapse`. This extra branch fires on every parent re-render where the prop hasn't changed, calling `setState` unnecessarily.

Consider simplifying to match `ObsGroupControl`:
```js
componentDidUpdate(prevProps) {
  if (this.props.collapse !== undefined && this.props.collapse !== prevProps.collapse) {
    this.setState({ collapse: this.props.collapse });
  }
}
```

File: src/components/bahmni-design-system/Section.jsx [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    const groupedRowControls = getGroupedControls(this.props.metadata.controls, 'row');
+    const disableClass = enabled ? '' : ' disabled';
+
+    const title = this.props.intl.formatMessage({
+      defaultMessage: label.value,
+      id: label.translationKey || 'defaultId',
+    });
+
+    return (
+      <div className="form-builder-fieldset">
```
Comment (ravinderkabli) [inline src/components/bahmni-design-system/Section.jsx]: suggestion: `ObsGroupControl` applies a `hidden` class via `classNames('form-builder-fieldset', { hidden: this.props.hidden })`, allowing form scripts to programmatically hide/show the control. Section hardcodes the class, so any `setHidden(true)` call on a Carbon Section will silently have no effect.

Fix:
```js
import classNames from 'classnames';
// add hidden: PropTypes.bool and hidden: false to defaultProps
<div className={classNames('form-builder-fieldset', { hidden: this.props.hidden })}>
```

**Review by ravinderkabli — CHANGES_REQUESTED** [review]
## Review for BAH-4641

Good progress on the Section migration — clean test suite and solid AC coverage. Two blocking issues to fix before merge.

Found 2 blocking issues and 2 suggestions.

**Acceptance Criteria**: 12/14 met (AC #6 missing edge-case test; AC #13 focus ring is a known issue acknowledged in the ticket)

File: src/components/bahmni-design-system/Section.jsx (hunk 1 above)
Comment (lingeswaranTW) [reply src/components/bahmni-design-system/Section.jsx:62]: Thanks for catching this — you're right that this.props.collapse stays stale after a user toggle, and ObsGroupControl does 
  use this.state.collapse in childProps. However, Section has an existing test that explicitly documents the opposite        
  contract:                                                                                                                  
                                                                                                                             
  ▎ "should pass prop collapse value to children regardless of internal toggle state"
                                                  
  The reason is that Section can contain nested Section components. If we pass this.state.collapse, a user toggling the      
  parent accordion overrides the independent collapse state of every child section — which is exactly the regression we saw
  (collapse child → collapse parent → expand parent → child gets force-expanded).                                            
                                          
  ObsGroupControl is safe with this.state.collapse because it doesn't nest other ObsGroupControl instances. Section is       
  different, so I've kept this.props.collapse for childProps to ensure programmatic "collapse all" cascades correctly while
  user-driven toggles stay local to the section that was clicked.

File: src/components/bahmni-design-system/CarbonContainer.jsx (hunk 2 above)
Comment (lingeswaranTW) [reply src/components/bahmni-design-system/CarbonContainer.jsx]: Got it resolved

File: src/components/bahmni-design-system/Section.jsx (hunk 3 above)
Comment (lingeswaranTW) [reply src/components/bahmni-design-system/Section.jsx]: Resolved

File: src/components/bahmni-design-system/Section.jsx (hunk 4 above)
Comment (lingeswaranTW) [reply src/components/bahmni-design-system/Section.jsx]: Resolved

**Review by ravinderkabli — APPROVED** [review]
Looks fine
